src/Common/src/System/IO/PersistedFiles.Unix.cs

```diff
@@ -94,5 +94,95 @@ private static void EnsureUserDirectories()
                 TopLevelHiddenDirectory,
                 SecondLevelDirectory);
         }
+
+        /// <summary>Gets the current user's home directory.</summary>
+        /// <returns>The path to the home directory, or null if it could not be determined.</returns>
+        internal static string GetHomeDirectory()
+        {
+            // First try to get the user's home directory from the HOME environment variable.
+            // This should work in most cases.
+            string userHomeDirectory = Environment.GetEnvironmentVariable("HOME");
+            if (!string.IsNullOrEmpty(userHomeDirectory))
+                return userHomeDirectory;
+
+            // In initialization conditions, however, the "HOME" enviroment variable may 
+            // not yet be set. For such cases, consult with the password entry.
+            unsafe
```

Comment:
Could there be an issue around memoization if the user changed their home directory? It looks like usermod can be used to change a home directory so there's the potential to return incorrect data if we save it off and then it changes